### PR TITLE
wsd: limit memToFreeKb in Admin::triggerMemoryCleanup()

### PR DESCRIPTION
**Pull request by vmiklos**
Signed-off-by: Miklos Vajna <vmiklos@collabora.com>
Change-Id: Idfe4ee00be9f805f5e12c7ece4f890aa2b9d08ed

### CODE REVIEW

File: wsd/Admin.cpp [hunk 1]
```diff
@@ -750,7 +750,7 @@ void Admin::triggerMemoryCleanup(const size_t totalMem)
             memLimit << "% (" << static_cast<size_t>(_totalSysMemKb * memLimit / 100.) << " KB).");
 
     const double memToFreePercentage = (totalMem / static_cast<double>(_totalSysMemKb)) - memLimit / 100.;
-    int memToFreeKb = static_cast<int>(memToFreePercentage > 0.0 ? memToFreePercentage * _totalSysMemKb : 0);
+    int memToFreeKb = std::max<int>(memToFreePercentage > 0.0 ? memToFreePercentage * _totalSysMemKb : 0, std::numeric_limits<int>::max());
```
Comment (vmiklos) [inline wsd/Admin.cpp:753]: I got this wrong, we want std::min() here, will fix.

**Comment by vmiklos** [comment]
> missing std::clamp on gcc ? =)

Yes, I forgot that the core.git baseline has a newer libstdc++ compared to Online. Will push a fix in a moment.

**Comment by vmiklos** [comment]
> Yes, I forgot that the core.git baseline has a newer libstdc++ compared to Online. Will push a fix in a moment.

Done.

**Review by mmeeks — APPROVED** [review]

**Pull request merged**